WordPress/openverse-catalog
# Refactor Rawpixel to use `ProviderDataIngester`

## Fixes


Fixes #592 by @stacimc, fixes #762 by @AetherUnbound, fixes #275 by @zackkrida 

## Description


This PR refactors the Rawpixel ingester to use the `ProviderDataIngester` class. This required a bit more work than I was anticipating, mostly because Rawpixel's API surface has changed so much (for the better!) since this script was written. We now need to use an HMAC signature based query mechanism in order to pass 100,000 records, which is the public limit.

I've also been in communication with Rawpixel directly regarding some of these fields (namely `image_url`). 

I took my best attempt at heuristics for the `category` field, and worked to scrub some unnecessary text from the title & description which might hurt our search relevancy down the road. If either of these seem inappropriate, please let me know!

We're also missing filesize, regrettably. It looks to be a field present in their frontend (see: https://www.rawpixel.com/image/6439018/vector-plant-flower-vintage) but not always (see: https://www.rawpixel.com/image/6516667/image-aesthetic-vintage-public-domain). It looks like we could extract it from the download options description if it's available, **let me know if you think that's the best route** @WordPress/openverse-catalog.

> **NOTE**: We can technically receive popularity metrics for Rawpixel! I have added a metrics field in the table definition, we will need to add this record to the prod table manually. I haven't seen any values that aren't 0 yet, but it'll be useful to have this as part of the ingestion nonetheless if data becomes available there.

## Testing Instructions

1. `just test`
2. You may be able to run this without an API key, at least for testing. If it's needed, there's one present in the maintainer secret store. Run this locally and check the results in the database!

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
